tracer/test/benchmarks/Benchmarks.Trace/AppSecBodyBenchmark.cs

```diff
@@ -0,0 +1,247 @@
+                            Name = "Europe",
+                            Planet = new Planet { Name = "Earth" }
+                        }
+                    }
+                },
+                IsHouse = true,
+                NameStreet = "lorem ipsum dolor sit amet"
+            },
+            Dogs = new List<Dog> {
+                    new Dog { Name = "toto", Dogs = new List<Dog> { new Dog { Name = "titi" }, new Dog { Name = "titi" } } },
+                    new Dog { Name = "toto", Dogs = new List<Dog> { new Dog { Name = "tata" }, new Dog { Name = "tata" } } },
+                    new Dog { Name = "tata", Dogs = new List<Dog> { new Dog { Name = "titi" }, new Dog { Name = "titi" }, new Dog { Name = "tutu" } } }
+                    }
+        };
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            security = Security.Instance;
+        }
+
+        [Benchmark]
+        public void AllCycleSimpleBody()
+        {
+#if NETFRAMEWORK
+            Debugger.Break();
+            using var ms = new MemoryStream();
+            using var sw = new StreamWriter(ms);
+            var httpContextMock = new HttpContext(new HttpRequest(string.Empty, string.Empty, string.Empty), new HttpResponse(sw));
+            security.InstrumentationGateway.RaiseBodyAvailable(httpContextMock, new Span(new SpanContext(1, 1), DateTimeOffset.UtcNow), new { });
+#else
+            var httpContextMock = new HttpContextMock(new HttpResponseMock());
+            security.InstrumentationGateway.RaiseBodyAvailable(httpContextMock, new Span(new SpanContext(1, 1), DateTimeOffset.UtcNow), new { });
+#endif
+        }
+
+        [Benchmark]
+        public void AllCycleMoreComplexBody()
+        {
+#if NETFRAMEWORK
+            using var ms = new MemoryStream();
+            using var sw = new StreamWriter(ms);
+            var httpContextMock = new HttpContext(new HttpRequest(string.Empty, string.Empty, string.Empty), new HttpResponse(sw));
+            security.InstrumentationGateway.RaiseBodyAvailable(httpContextMock, new Span(new SpanContext(1, 1), DateTimeOffset.UtcNow), complexModel);
+#else
+            var httpContextMock = new HttpContextMock(new HttpResponseMock());
+            security.InstrumentationGateway.RaiseBodyAvailable(httpContextMock, new Span(new SpanContext(1, 1), DateTimeOffset.UtcNow), complexModel);
+#endif
+        }
+
+        [Benchmark]
+        public void BodyExtractorSimpleBody() => BodyExtractor.GetKeysAndValues(new { });
+
+        [Benchmark]
+        public void BodyExtractorMoreComplexBody() => BodyExtractor.GetKeysAndValues(complexModel);
+    }
+
+#if !NETFRAMEWORK
+    public class HttpContextMock : HttpContext
+    {
+        private readonly HttpResponseMock responseMock;
+
+        public HttpContextMock(HttpResponseMock responseMock)
+        {
+            this.responseMock = responseMock;
+        }
+        public override ConnectionInfo Connection => throw new NotImplementedException();
```

Comment:
I'm not sure all this is necessary? I think you can use `DefaultHttpContext`?

Reply:
yes I wasnt sure how to make sure the response was disposing the waf pointers, but with disposing them manually now, no need for all this anymore thanks